src/app/input-bar/input-bar.component.ts

```diff
@@ -37,46 +39,120 @@ export class InputBarComponent implements OnInit {
 
   private parseInput(input: string) {
     const filters: FilterOption<any>[] = [];
-    let groupBy: GroupByKeys = GroupByKeys.None;
+    let grouping: GroupingOptions = {
+      groupBy: GroupByKeys.None,
+      sort: GroupOrders.None
+    };
 
     if (!input) {
       return {
         filters,
-        groupBy
+        grouping
       };
     }
-    const splitInput = input.trim().split(' ');
 
-    for (let i = 0; i < splitInput.length; i++) {
-      if (splitInput[i] !== '{') {
-        const newFilter = this.parseArgument(splitInput[i]);
+    // We put each space seperated token in a stack
+    const stackInput = input.trim().split(' ').reverse();
+
+    let currInputArg: string;
+    while (stackInput.length > 0) {
+      currInputArg = stackInput.pop();
+      if (currInputArg === '{') {
+        grouping = this.parseGrouping(stackInput);
+        if (grouping.groupBy === GroupByKeys.Rewrite) {
+          // An empty inputString tells us to not highlight this regex filter
+          filters.push({
+            filterKey: FilterKeys.Regex,
+            inputString: '',
+            value: grouping.regex,
+            isNegation: false,
+            isValidInput: true
+          });
+        }
+      } else {
+        const newFilter = this.parseArgument(currInputArg);
         if (newFilter.isValidInput) {
           filters.push(newFilter);
         }
-      } else if (i + 2 >= splitInput.length || splitInput[i + 2] !== '}') {
-        // To group we require the syntax '{ grouping_key }'
-        console.error('Invalid input, to group links make sure there are ' +
-            'spaces between the brackets and the grouping key.');
-        break;
+      }
+    }
+
+    return {
+      filters,
+      grouping
+    };
+  }
+
+  /**
+   * Takes a stack of inputs starting immediately after the opening bracket and
+   * parses each one, incoroprating it into a GroupingOptions object until it
+   * finds the closing bracket.
+   * @param { string[] } stackInput The input tokens as a stack of strings
+   * starting right after the opening bracket.
+   * @return { GroupingOptions } The constructed GroupingOptions object.
+   */
+  private parseGrouping(stackInput: string[]): GroupingOptions {
+    const grouping: GroupingOptions = {
+      groupBy: GroupByKeys.None,
+      sort: GroupOrders.None
+    };
+
+    let currGroupArg: string;
+    while (stackInput.length > 0) {
+      currGroupArg = stackInput.pop();
+      if (currGroupArg === '}') {
+        return grouping;
+      } else if (currGroupArg.startsWith(GroupingModifiers.Sort)) {
+        currGroupArg = currGroupArg.substring(GroupingModifiers.Sort.length);
+        switch (currGroupArg) {
```

Comment:
Okay!

Reply:
Let me rephrase my previous comment.

I think we should *only* support the following

```
lex-asc
lex-desc
size-asc
size-desc
```

instead of supporting multiple alternatives. 

It's not obvious for user that `ascending` or `asc` is treated as `lexico-asc` using this syntax. It's also not obvious `size` means the same thing as `size-asc`.

Reply:
Yeah, that's why I have it set up so that if none of the "officially supported" syntax it will replace it with the official syntax in the input bar to correct the user so that if they input asc, they will see that the parser is treating that as if it were lex-asc. That is what this line does (newInput is the variable bonded to input bar) :

`this.newInput = this.newInput.replace(original, 'sort:lex-desc');`

I though this was the behavior that you described above, let me know if I misunderstood.

Reply:
Oh you get it correctly. My idea was to do this transformation before "consumeAndParseGrouping()". I wanted to simplify this function a bit since it's doing many things at once. But after looking it a bit more, I think it might complicate things further.

I guess we can keep this logic then